structure/structure.py

```diff
@@ -15,24 +15,66 @@
 from griptape.tools import Calculator
 
 
-# Returns True if this program is being run from within Griptape Cloud or via the Skatepark emulator
-def is_running_in_managed_environment():
+def is_running_in_managed_environment() -> bool:
+    """Determine if the program is running in a managed environment (e.g., Griptape Cloud or Skatepark emulator).
+    Returns:
+        bool: True if the program is running in a managed environment, False otherwise.
+    """
     return "GT_CLOUD_RUN_ID" in os.environ
 
 
+def get_listener_url() -> str:
+    """The event driver requires a URL, stored in the GT_CLOUD_BASE_URL 
+    environment variable.
+    In Griptape Cloud, this environment variable is provided for you.
+    In Skatepark, you may optionally provide your own if you have overridden
+    the default URL and port that the emulator is listening on. Otherwise,
+    it will use the default URL and port.
+
+    Returns:
+        The URL of the host listening to the Structure.
+    """
+    return os.environ.get("GT_CLOUD_BASE_URL")
+
+
+def get_listener_api_key() -> str:
+    """The event driver expects a Griptape API Key as a parameter.
+    When your program is running in Griptape Cloud, you will need to provide a 
+    valid Griptape API Key in the GT_CLOUD_API_KEY environment variable 
+    in order to authorize calls. 
+    You can create an API Key by visiting https://cloud.griptape.ai/keys 
+    When running in Skatepark, the API key is not needed since it isn't validating calls.
+
+    Returns:
+        The Griptape API Key to authorize calls.
+    """
+    api_key = os.environ.get("GT_CLOUD_API_KEY")
+    if is_running_in_managed_environment() and not api_key:
+        print("""
+              ****WARNING****: No value was found for the 'GT_CLOUD_API_KEY' environment variable.
+              This environment variable is required when running in Griptape Cloud for authorization.
+              You can generate a Griptape API Key by visiting https://cloud.griptape.ai/keys.
+              Specify it as an environment variable when creating a Managed Structure in Griptape Cloud.
+              """
+              )
+    return api_key
+
+
 # If running completely local, such as within an IDE, load environment vars.
 if not is_running_in_managed_environment():
     load_dotenv()
 
 input = sys.argv[1]
 
-# We need an event driver to communicate events from our program back to Skatepark
-listener_base = os.environ.get("GRIPTAPE_SKATEPARK_URL_OVERRIDE", "http://127.0.0.1")
-listener_port = os.environ.get("GRIPTAPE_SKATEPARK_PORT_OVERRIDE", "5000")
-listener_url = f"{listener_base}:{listener_port}"
-
+# We need an event driver to communicate events from our program back to our
+# host, which could be the locally-run Skatepark or Griptape Cloud.
+# The event driver requires a URL, stored in the GT_CLOUD_BASE_URL environment variable.
```

Comment:
This reads a bit like it's _required_ for it to be in `GT_CLOUD_BASE_URL`. We read from there by default, but users can store it wherever they'd like and then manually pass it in.

Reply:
ah yeah, poor choice of tone. Will update.